src/tests/roc_audio/test_profiler.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <CppUTest/TestHarness.h>
+
+#include "roc_audio/profiler.h"
+#include "roc_core/heap_allocator.h"
+
+namespace roc {
+namespace audio {
+
+namespace {
+
+const core::nanoseconds_t interval = 1000;
+const int sample_rate = 300;
+const int num_channels = 1;
+core::HeapAllocator allocator;
+
+} // namespace
+
+TEST_GROUP(profiler) {};
+
+TEST(profiler, test_moving_average) {
```

Comment:
This test, covers every aspect of the algorithm yet it isn't very readable and looks like I have just picked magic values for the test. Is there any better way to test this and should I write any other tests for this algorithm?

Reply:
I think the approach is OK, but it would be better to replace magic constant and to replace expected values with an expression using those constants. So e.g. you're passing Speed1 and Speed2 and expecting average (Speed1 + Speed2) / 2.